File: tests/fixtures/privacy-scanner-probes/comment_probe.py
"""스캐너 음성 대조용 합성 파일 (파이썬 문법). 제품 코드가 아니다.

`#`는 `.py`에서만 줄 주석이므로 같은 결함을 파이썬 쪽에서도 따로 확인한다 —
한쪽만 고치면 다른 쪽이 남는다.

`SCAN_ROOTS` 밖(`tests/fixtures/`)에 있어 전수 스캔이 자기 탐침을 잡지 않는다
(그 배치 자체를 `test_합성_탐침이_스캔_루트_밖에_있다`가 단언한다).

**실행되는 코드가 아니지만 유효한 코드로 쓴다.** `ruff`·`mypy`가 이 파일도 검사하므로,
문법을 어겨 두면 그 게이트를 통과시키려고 결국 예외 경로를 파게 된다 — 탐침 하나 때문에
검사 범위에 구멍을 내는 것이 이 저장소가 반복해서 거부해 온 형태다.
"""

import logging
from dataclasses import dataclass

logger = logging.getLogger(__name__)


@dataclass
class _Stats:
    count: int


@dataclass
class _Draft:
    """본문(`value`)과 집계(`stats`)를 함께 든 값 — 실물 `ConversionResult`의 모양이다."""

    stats: _Stats
    value: str


def 줄주석_닫는괄호(draft: _Draft) -> None:
    logger.info(
        "완료 %s %s",
        draft.stats.count,  # 건수) 설명
        draft.value,
    )


def 주석없음(draft: _Draft) -> None:
    logger.info(
        "완료 %s %s",
        draft.stats.count,
        draft.value,
    )

File: tests/fixtures/privacy-scanner-probes/CommentProbe.kt
package probes

// 스캐너 음성 대조용 **합성 파일**. 제품 코드가 아니다.
//
// codex stop-time 게이트가 찾은 결함 — 인자 파서가 주석을 모른다.
//
// 재현 조건 셋이 **함께** 성립해야 샌다:
//   ① 주석 안에 짝 없는 `)` 가 있다  → 인자 구간이 거기서 조기에 닫힌다
//   ② 그 앞에 **안전한** 본문 접근이 있다 → refine 이 "찾았고 전부 안전" 경로로 들어간다
//   ③ 진짜 위험한 접근이 그 뒤에 있다   → 잘린 구간 밖이라 아예 검사되지 않는다
//
// ②가 없으면 refine 이 "본문 이름 없음"으로 보수적 CAUGHT 를 내므로 드러나지 않는다.
// 그래서 "주석이 있으면 샌다"가 아니라 **"안전한 접근이 방패가 된다"**가 정확한 서술이다.
object CommentProbe {
    fun 줄주석_닫는괄호(draft: Any) {
        logger.info(
            "완료 {} {}",
            draft.stats.count, // 건수) 설명
            draft.value,
        )
    }

    fun 블록주석_닫는괄호(draft: Any) {
        logger.info(
            "완료 {} {}",
            draft.stats.count, /* 건수) 설명 */
            draft.value,
        )
    }

    // 대조군 — 같은 코드에서 주석만 뺐다. 주석 유무로 검출이 갈리면 안 된다.
    fun 주석없음(draft: Any) {
        logger.info(
            "완료 {} {}",
            draft.stats.count,
            draft.value,
        )
    }
}
